tests/JsonParseErrorExceptionTest.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/*
+    Copyright 2014 Rustici Software
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+use TinCan\JsonParseErrorException;
+
+class JsonParseErrorExceptionTest extends PHPUnit_Framework_TestCase
+{
+
+    private $exception;
+
+    private $malformedValue  = '.....';
+    private $jsonErrorNumber;
+    private $jsonErrorMessage;
+
+    public function setUp()
+    {
+        $this->jsonErrorNumber = JSON_ERROR_SYNTAX;
+        $this->jsonErrorMessage = 'Syntax error, malformed JSON';
+        $this->exception = new JsonParseErrorException(
+            $this->malformedValue,
+            $this->jsonErrorNumber,
+            $this->jsonErrorMessage
+        );
+    }
+
+    public function testThatWeCanGetOurInformationBackOut()
+    {
+        $this->assertEquals($this->malformedValue, $this->exception->malformedValue());
+        $this->assertEquals($this->jsonErrorNumber, $this->exception->jsonErrorNumber());
+        $this->assertEquals($this->jsonErrorMessage, $this->exception->jsonErrorMessage());
+    }
+
+    public function testThatWeGetAMeaningfulErrorMessage()
```

Comment:
Its fine to be inconsistent here given that there is a demonstrated improvement. I'm not breaking consistency for the sake of it.

The potential impact is that the code will look alien in comparison to what is already there. Meaning that at first glance it is more _difficult to understand_ because it doesn't follow a pattern. Given that this is (in my view) _easier to understand_ it solves the same problem it creates by not conforming.